tests/manage/pv_services/test_resource_deletion_during_pvc_pod_creation_and_io.py

```diff
@@ -133,41 +219,41 @@ def operations_base(self, resource_to_delete):
         # Getting result of PVC creation as list of PVC objects
         pvc_objs_new = bulk_pvc_create.result()
 
-        # Updating self.pvc_objs_new for the purpose of teardown
-        self.pvc_objs_new.extend(pvc_objs_new)
-
-        # Verify PVCs are Bound
+        # Confirm PVCs are Bound
         for pvc_obj in pvc_objs_new:
-            assert pvc_obj.ocp.wait_for_resource(
-                condition=constants.STATUS_BOUND, resource_name=pvc_obj.name,
-                timeout=240, sleep=10
-            ), (
-                f"Wait timeout: PVC {pvc_obj.name} is not in 'Bound' status"
+            helpers.wait_for_resource_state(
+                resource=pvc_obj, state=constants.STATUS_BOUND, timeout=120
             )
+            pvc_obj.reload()
         log.info("Verified: New PVCs are Bound.")
 
         # Getting result of pods creation as list of Pod objects
         pod_objs_new = bulk_pod_create.result()
 
-        # Updating self.pod_objs for the purpose of teardown
-        self.pod_objs.extend(pod_objs_new)
-
         # Verify new pods are Running
         for pod_obj in pod_objs_new:
-            assert pod_obj.ocp.wait_for_resource(
-                condition=constants.STATUS_RUNNING,
-                resource_name=pod_obj.name, timeout=240, sleep=10
-            ), (
-                f"Wait timeout: Pod {pod_obj.name} is not in 'Running' "
-                f"state even after 120 seconds."
+            helpers.wait_for_resource_state(
+                resource=pod_obj, state=constants.STATUS_RUNNING
             )
-        log.info("Verified: All pods are Running.")
+            pod_obj.reload()
+        log.info("Verified: All new pods are Running.")
 
         # Verify IO
-        log.info("Fetching IO results.")
+        log.info("Fetching IO results from IO pods.")
         for pod_obj in io_pods:
-            get_fio_rw_iops(pod_obj)
-        log.info("Verified IO result on pods.")
+            fio_result = pod_obj.get_fio_results()
+            log.info(f"IOPs after FIO on pod {pod_obj.name}:")
+            log.info(
+                f"Read: {fio_result.get('jobs')[0].get('read').get('iops')}"
+            )
+            log.info(
+                f"Write: {fio_result.get('jobs')[0].get('write').get('iops')}"
+            )
+            err_count = fio_result.get('jobs')[0].get('error')
```

Comment:
We can check for fio errors before logging Read and Write iops results.
Also can we have the error checking in the  pod.get_fio_rw_iops() itself?

Reply:
> We can check for fio errors before logging Read and Write iops results.
Sure, I will modify that.
> Also can we have the error checking in the pod.get_fio_rw_iops() itself?
get_fio_rw_iops is used in many tests to log fio result. If error checking is done there, the purpose of the function will change. Also, function name should be changed appropriately.
Should we consider modifying the same function for this purpose ?.

Reply:
> We can check for fio errors before logging Read and Write iops results.

Done

Reply:
```get_fio_rw_iops is used in many tests to log fio result. If error checking is done there, the purpose of the function will change```
ack